cypress/integration/user-flow.spec.js

```diff
@@ -27,6 +36,34 @@ describe('Access to the new metadata app', () => {
     cy.get('.navsec').should('not.exist');
     cy.url().should('include', '/user/login');
     cy.url().should('include', 'came_from=%2Fdataset%2Fnew-metadata');
+    // Same for edit dataset page
+    cy.visit('/dataset/edit-new/dataset-id', { failOnStatusCode: false });
+    cy.get('.navsec').should('not.exist');
+    cy.url().should('include', '/user/login');
+    cy.url().should('include', 'came_from=%2Fdataset%2Fedit-new%2Fdataset-id');
+  });
+
+  it('Returns 404 when trying to edit non-existing dataset', () => {
+    cy.login();
+    cy.visit('/dataset/edit-new/dataset-id', { failOnStatusCode: false });
+    cy.url().should('include', '/dataset/edit-new/dataset-id');
+    cy.contains('404 Not Found');
+  });
+});
+
+describe('Deleting a dataset', () => {
+  it('Has "Delete" button', () => {
+    cy.login();
+    cy.visit('/dataset/test-dataset-1');
+    cy.get('.btn-danger').should('exist').contains('Delete');
+  });
+
+  it('Displays confirmation page when clicked', () => {
+    cy.login();
+    cy.visit('/dataset/test-dataset-1');
+    cy.get('.btn-danger').click();
+    cy.contains('Are you sure you want to delete dataset - test-dataset-1?');
+    cy.get('.btn-primary').contains('Confirm Delete');
```

Comment:
Maybe we need also a test for checking if it's able to actually delete dataset? Let's say deleting it and then trying to access through the url